### ISSUE TITLE
library: uninformative error messages when referencing image without tag when latest tag does not exist

### ISSUE THREAD

**Issue by gmkurtzer** [apptainer/singularity#2722]
## Version of Singularity:

```
$ singularity version
3.0.3-756.ga72e4274
```

### Expected behavior

"latest" tag isn't assumed and it is confusing when pulling a container that doesn't have latest tag.

### Actual behavior

I see why this behavior maybe deemed as appropriate, but from a user's perspective the error is confusing and unexpected.

### Steps to reproduce behavior

```
$ singularity search centos
No users found for 'centos'

No collections found for 'centos'

Found 7 containers for 'centos'
	library://dtrudg/linux/centos
		Tags: 6 7 centos6 centos7 latest
	library://library/default/centos
		Tags: 6 7 latest
	library://gmk/demo/centos-vim
		Tags: latest
	library://mroche/baseline/centos
		Tags: 7 7.5 7.5.1804 7.6 7.6.1810 latest
	library://gmk/default/centos7-devel
		Tags: latest
	library://emmeff/default/centos7-python36
		Tags: 1.0
	library://westleyk/official/centos
		Tags: 7

$ singularity pull library://emmeff/default/centos7-python36
FATAL:   While getting image info: the requested image was not found in the library
$ singularity pull library://emmeff/default/centos7-python36:1.0
INFO:    Downloading library image
 82.90 MiB / 82.90 MiB [==========================================================================================================] 100.00% 10.03 MiB/s 8s
$
```

We either need to change the error to include a tag, show the available tags, or assume latest.

**Comment by bauerm97** [comment]
@gmkurtzer So to start, I _do_ think it's necessary for us to add the `latest` tag to an image which has no tag. If we don't add a tag, we have no way of pulling an image. You could argue that we should just grab the tag with the most recent timestamp, but I'm not sure the library supports this, and it would be confusing in the case of e.g. a nightly devel build being tagged, but a `latest` tag existing to mark the latest **stable** version of an image. 

I think this primarily boils down to a logging issue. Let's make sure to do the following:

1. When a tag is not specified, let's output a nice message saying something along the lines of "No tag specified, searching for tag: latest".
2. When no latest tag is then found, we print out a much more descriptive error message. The error should detail if an image collection exists, and if it does exist but without the tags it should also note that. Something like "The image [...] does not have the specified tag: latest", vs "The image [...] does not exist" (in cases where the image simply does not exist, regardless of tag)

I think taking those two steps will turn this into an actionable item and solve your concern here.

**Comment by gmkurtzer** [comment]
If we aren't going to tag images automatically with latest, then I'd suggest that we error when pushing an image with no tag with a helpful suggestion.

**Comment by jscook2345** [comment]
@WestleyK @ikaneshiro @bauerm97 

What release do you want to target this for?

**Comment by GodloveD** [comment]
The immediate fix here is to give a more verbose error message that explicitly states it is looking for the latest tag.  

Long-term we probably want to make sure that we are doing exactly what Docker Hub is doing as far as automatically tagging newly pushed containers with `latest` too.

**Comment by GodloveD** [comment]
This issue was fixed(ish) with #3329.  If we want to revisit how we are handling `latest` I think a new issue should be opened.
